src/util.coffee

```diff
@@ -13,10 +14,24 @@ one_day = 1000 * 60 * 60 * 24
 emojiList = []
 
 module.exports = (robot) ->
-  ensure_no_public_tokens: (request, token) ->
+  ensure_no_public_tokens: (request, authJsonString) ->
     if request.message.room[0] == 'C'
-      slack.chat.delete({token: token, channel: request.message.room, ts: request.message.id})
-      request.send("Don't go posting slack auth tokens in public channels ya dummy. Delete that or I'm telling mom.")
+      try
+        authJson = JSON.parse(authJsonString)
+        token = authJson["token"]
+        cookie = authJson["cookie"]
+        subdomain = authJson["domain"] || authJson["subdomain"] || request.message.user.slack.team_id.replace(/:/g,'').trim()
+        slack = new SlackClient(subdomain, cookie)
+        if token && cookie
```

Comment:
I think the pedantic swe in me wants to say there's a way to invert this conditional tree to `fail fast`, e.g. 
```
if !(token && cookie)
    fail message
parse out (response) ->
    delete and chastise
    return
tell mom
```

that said i _do not care_ and _hate software_

Reply:
It was kinda painful writing the same message for every fail path so I like your idea much more - I can make a change for this